components/layout_2020/display_list/mod.rs

```diff
@@ -450,7 +451,65 @@ impl Fragment {
             self.build_display_list_for_text_decoration(fragment, builder, &rect, &color);
         }
 
-        // Text.
+        if let Some(range) = fragment.selection_range {
+            let baseline_origin = rect.origin;
+            let start = glyphs_advance_by_index(
+                &fragment.glyphs,
+                range.begin(),
+                baseline_origin,
+                fragment.justification_adjustment,
+            );
+
+            let end = glyphs_advance_by_index(
+                &fragment.glyphs,
+                range.end(),
+                baseline_origin,
+                fragment.justification_adjustment,
+            );
+
+            let selection_rect = LayoutRect::new(
```

Comment:
This won't work with vertical text, so better put a TODO here.

And does this work when `start` is larger than `end` (due to eg reversed selection or RTL)?